gobblin-data-management/src/main/java/gobblin/data/management/retention/action/MultiAccessControlAction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.data.management.retention.action;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.hadoop.fs.FileSystem;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigRenderOptions;
+
+import gobblin.data.management.retention.dataset.ConfigurableCleanableDataset;
+import gobblin.data.management.version.DatasetVersion;
+import gobblin.util.ConfigUtils;
+
+
+/**
+ * A wrapped around {@link AccessControlAction} that delegates the {@link #execute(List)} call to multiple embedded
+ * {@link AccessControlAction}s
+ * Use {@link MultiAccessControlActionFactory} to create new {@link MultiAccessControlAction}s
+ */
+public class MultiAccessControlAction extends RetentionAction {
```

Comment:
A dataset can have multiple access control policies for its versions. For instance below I have shown `restrictedAccess` for a few versions and `noAccess` for a few others. For the below case we create two `AccessControlAction`s (Note that an `AccessControlAction` has a `SelectionPolicy` and permissions to set for this selection). 
The `MultiAccessControlAction` is a container for these two real access control actions.

<pre>
accessControl {
           
            ### Read by the MultiAccessControlAction and creates a AccessControlAction for each.
            policies = [restricted,noAccess]
            restricted {
                 selection {
                    policy.class=SelectionBetweenTimeBasedSelectionPolicy
                    timeBased.maxLookbackTime=3y
                    timeBased.minLookbackTime=2y
                 }
                 mode : 770
                  user : u1
                  group : g1
              }
             noAccess {
                 selection {
                    policy.class=SelectionBetweenTimeBasedSelectionPolicy
                    timeBased.maxLookbackTime=5y
                    timeBased.minLookbackTime=3y
                 }
                 mode : 700
                 user : u2
                 group : g2
              }
         }
</pre>